portal/static/js/main.js

```diff
@@ -1894,30 +1894,37 @@ var assembleContent = {
                 };
             };
 
-            var studyId = $("#profileStudyId").val();
-            var siteId = $("#profileSiteId").val();
+            var studyIdField = $("#profileStudyId");
+            var siteIdField = $("#profileSiteId");
+            var hasStudyId = studyIdField.length > 0 && studyIdField.is(":visible");
+            var hasSiteId = siteIdField.length > 0 && siteIdField.is(":visible");
+            var studyId = studyIdField.val();
+            var siteId = siteIdField.val();
 
-            if (hasValue(studyId) || hasValue(siteId)) {
+
+            if (hasStudyId || hasSiteId) {
                 var identifiers = null;
                 //get current identifier(s)
                 $.ajax ({
                     type: "GET",
-                    url: '/api/demographics/'+userId,
+                    url: "/api/demographics/"+userId,
                     async: false
                 }).done(function(data) {
                     if (data && data.identifier) {
                         identifiers = [];
                         (data.identifier).forEach(function(identifier) {
                             if (identifier.system != SYSTEM_IDENTIFIER_ENUM["external_study_id"] &&
                                 identifier.system != SYSTEM_IDENTIFIER_ENUM["external_site_id"] &&
-                                identifier.system != SYSTEM_IDENTIFIER_ENUM["practice_region"]) identifiers.push(identifier);
+                                identifier.system != SYSTEM_IDENTIFIER_ENUM["practice_region"]) {
+                                identifiers.push(identifier);
+                            }
                         });
-                    };
-                }).fail(function() {
-                   // console.log("Problem retrieving data from server.");
+                    }
+                }).fail(function(xhr) {
+                   tnthAjax.reportError(userId, "api/demographics"+userId, xhr.responseText);
```

Comment:
Does `tnthAjax.reportError` log to the frontend console as well? I can see error reporting being useful in both places :)

Reply:
added, ivan!